### ISSUE TITLE
Generate error when tsconfig sets "paths" but not "baseUrl"

### ISSUE THREAD

**Issue by mjbvz** [Microsoft/TypeScript#21552]
**TypeScript Version:**  2.7.1

**Search Terms:** 
- tsconfig
- jsconfig
- paths
- baseurl

**Repo**
Create a `tsconfig.json`:

```json
{
     "compilerOptions": {
        "paths": {
            "@app/*": ["app/*"]
        }
    }
}
```

**Expected behavior:**
Should show diagnostic in the `tsconfig` since we set `paths` but not `baseUrl`

**Actual behavior:**
No diagnostic reported

**Comment by MagWeaver** [comment]
Can I take this one?

**Comment by mhegazy** [comment]
go for it. [`verifyCompilerOptions`](https://github.com/Microsoft/TypeScript/blob/master/src/compiler/program.ts#L2050) would be where i expect a check to be.

**Comment by MagWeaver** [comment]
I installed a local build of TS and used the following `tsconfig.json` file.

```{
"compilerOptions": {
        "paths": {
            "@app/*": ["app/*"]
        }
    }
}
```
I ran `node tsc.js` and get the diagnostic:
`tsconfig.json(3,9): error TS5060: Option 'paths' cannot be used without specifying '--baseUrl' option.`

Running `node tsc.js test.js` does not return the diagnostic. This is consistent with the description of tsconfig.json http://www.typescriptlang.org/docs/handbook/tsconfig-json.html. 

> Using tsconfig.json
> - By invoking tsc with no input files, in which case the compiler searches for the tsconfig.json file starting in the current directory and continuing up the parent directory chain.
> - By invoking tsc with no input files and a --project (or just -p) command line option that specifies the path of a directory containing a tsconfig.json file, or a path to a valid .json file containing the configurations.
> When input files are specified on the command line, tsconfig.json files are ignored.

**Comment by MagWeaver** [comment]
@mjbvz @mhegazy 
Does my description make sense or am I missing something? Doesn't seem like there is an issue here.

**Comment by mhegazy** [comment]
thanks! closing.
